source/Calamari/Kubernetes/Helm/HelmTemplateValueSourcesParser.cs

```diff
@@ -20,6 +20,33 @@ public static IEnumerable<string> ParseTemplateValuesSources(RunningDeployment d
 
             var parsedJsonArray = JArray.Parse(templateValueSources);
 
+            return ParseFilenamesFromTemplateValuesArray(deployment, fileSystem, log, parsedJsonArray);
+        }
+
+        public static IEnumerable<string> ParseTemplateValuesFilesFromDependencies(RunningDeployment deployment, ICalamariFileSystem fileSystem, ILog log)
+        {
+            var templateValueSources = deployment.Variables.Get(SpecialVariables.Helm.TemplateValuesSources);
+
+            if (string.IsNullOrWhiteSpace(templateValueSources))
+                return Enumerable.Empty<string>();
+
+            var parsedJsonArray = JArray.Parse(templateValueSources);
+
+            //we are only interested in the values files in external dependencies (chart/package/git repo), so filter this array
+            var relevantTypes = parsedJsonArray.Where(t =>
+                                                      {
+                                                          var type = (TemplateValuesSourceType)Enum.Parse(typeof(TemplateValuesSourceType),t.Value<string>(nameof(TemplateValuesSource.Type)));
+                                                          return type == TemplateValuesSourceType.Chart || type == TemplateValuesSourceType.Package || type == TemplateValuesSourceType.GitRepository;
+                                                      })
+                                               .ToList();
```

Comment:
We take the same variable and just filter the types to those that we need to perform variable substitution on.